assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -96,13 +123,119 @@ const actions = {
 	dismissLostConversionReportingEvents() {
 		return dismissLostConversionReportingEventsStore.actions.fetchDismissLostConversionReportingEvents();
 	},
+
+	/**
+	 * Stores conversion reporting inline data in the datastore.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} data Inline data, usually supplied via a global variable from PHP.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveConversionReportingInlineData( data ) {
+		invariant( data, 'data is required.' );
+
+		return {
+			payload: { data },
+			type: RECEIVE_CONVERSION_REPORTING_INLINE_DATA,
+		};
+	},
+};
+
+export const reducer = createReducer( ( state, { payload, type } ) => {
+	switch ( type ) {
+		case RECEIVE_CONVERSION_REPORTING_INLINE_DATA: {
+			const { newEvents, lostEvents } = payload.data;
+
+			return {
+				...state,
+				detectedEventsChange: {
+					newEvents,
+					lostEvents,
+				},
+			};
+		}
+
+		default: {
+			return state;
+		}
```

Comment:
Sorry, I should have clarified the implications of this suggestion which is that the reducer can be significantly simplified using this approach because `state` is like a draft of the new state, so it is safe to mutate and does not even need to be returned. You can see how it's used here in its tests https://github.com/google/site-kit-wp/pull/5800/files

For example though, the action here can be updated like
```js
state.detectedEventsChange = { newEvents, lostEvents };
```
Normally this would be against proper reducer rules as it would not be a pure function, but using Immer, it is :)

Reply:
Ah I see, thanks. It is indeed simpler :)

PR updated